### DISCUSSION TITLE
Docker Mods

### DISCUSSION THREAD

**Discussion by Protektor-Desura** [RandomNinjaAtk/docker-lidarr-extended#20] [Ideas]
Have you ever thought about making this a docker mod instead? The reason I ask is because it would seem to me that it could unhook your release schedule from Lidarr. You could do your updates at your own schedule instead of having to do a release everything time Lidarr is updated.

I am currently trying to switch all my Sonarr & Radarr setups to use SMA as a docker mod. What this does is allow a better modular setup. Like SMA needs MESA 21+ in order to properly transcode to HEVC under VAAPI. So if you don't do that INSIDE the container then you won't get HEVC encoding with VAAPI which is needed for AMD & Intel APU/GPUs.

So I'm using the SMA mod and the Jellyfin-AMD mod and you could even use an ffmpeg mod to make sure you have the most up to date ffmpeg as well. While some are still using ffmpeg 4, you could instead pick ffmpeg4 or ffmpeg5, just by picking which ffmpeg mod you load.

In this case you could basically have:
a AMVD MOD
a deezer/deemix MOD
a tidal/streamrip MOD
a beets MOD

Then you just add whatever Enviroment variables need for each mod to your docker.

I don't know if this would increase or reduce your support issues, but it might be more flexible for your release and development schedule though.

**Comment by RandomNinjaAtk** [comment]
I'm happy with the current implementation, and not planning on re-designing things... However, if it doesn't require me to create docker builds using docker hub, than maybe I'd consider because that would reduce my expenses. But if I still need to have docker hub to do the builds, then its not worth re-designing something that is already working as intended.

**Comment by Protektor-Desura** [comment]
You do know that GitHub has it's own docker repository right? ghcr.io

Also as a side note for your sonarr-sma & radarr-sma dockers. If you don't install Mesa 21+ INSIDE the docker then you won't be able to use VAAPI to do H.265/HEVC encoding. I currently use linuxserver jellyfin-AMD docker mode. It is required for VAAPI which is what you use typically for both Intel & AMD GPU transcoding.
https://mods.linuxserver.io/?mod=jellyfin
https://github.com/linuxserver/docker-mods/tree/jellyfin-amd

Use it like this:
DOCKER_MODS=linuxserver/mods:jellyfin-amd

**Comment by RandomNinjaAtk** [comment]
I'm not going to redesign what is already working, also from quick review, it does not appear that github builds the images, it only stores them, and there is also a cost associated with it. So it's not any better...
